### DISCUSSION TITLE
Where's the "Prompting Expert Mode"?

### DISCUSSION THREAD

**Discussion by ChiNoel-osu** [langgenius/dify#4437] [Help] [answered]
I was just started using Dify and I saw that there's this "Prompting Expert Mode" [here](https://docs.dify.ai/features/prompt-engineering), but I can't find it anywhere in the latest version of Dify. And after a quick search I found out that the expert mode is removed.
I want a way to implement my own raw prompt template, not just some high-level `USER` and `ASSISTANT` message. I also want to use some of the more complicated prompt formats like `ChatML` which looks like this:
```
<|im_start|>system 
[SYSTEM PROMPT GOES HERE]
<|im_end|> 
<|im_start|>user 
[USER QUERY GOES HERE]
<|im_end|> 
<|im_start|>assistant 
[ASSISTANT RESPONSE]
<|im_end|> 
<|im_start|>user 
[USER SECOND QUERY]
<|im_end|> 
<|im_start|>assistant 
[ASSISTANT RESPONSE]
....
```
Any help is appreciated.

**Comment by patryk20120** [comment]
You can use LLM block in chatflows and workflows. In LMM block you can implement SYSTEM / USER and ASSISTANT prompts

**Reply by ChiNoel-osu** [reply, marked as answer]
Thank you for the reply. Yes, the LLM block allows input for those prompts but they're high-level implementations. I want to have direct control over the raw prompt that will be send to the LLM.
I researched for a bit and it looks like Dify just makes API calls to inference backends and probably don't have control over those. I'm using Xinference, I guess I need to find out how to set the prompt template on there.

**Reply by patryk20120** [reply]
Well, this what you set in LLM block is send over by API call to the model provider.
